applications/GeoMechanicsApplication/custom_strategies/strategies/geo_mechanics_newton_raphson_erosion_process_strategy.hpp

```diff
@@ -425,5 +398,52 @@ class GeoMechanicsNewtonRaphsonErosionProcessStrategy
             }
         }
     }
+
+    template <typename FilteredElementType>
+    void DetermineOpenPipingElements(const FilteredElementType& rPipingElements, double MaxPipeHeight)
+    {
+        bool grow                      = true;
+        auto number_of_piping_elements = rPipingElements.size();
+        auto number_of_open_piping_elements = this->InitialiseNumActivePipeElements(rPipingElements);
+        while (grow && (number_of_open_piping_elements < number_of_piping_elements)) {
+            KRATOS_INFO("DetermineOpenPipingElements")
+                << "grow " << grow << "number_of_open_piping_elements" << number_of_open_piping_elements
+                << " number_of_piping_elements " << number_of_piping_elements << std::endl;
+
+            // get tip element and activate
+            auto tip_element = rPipingElements.at(number_of_open_piping_elements);
+            number_of_open_piping_elements += 1;
+            tip_element->SetValue(PIPE_ACTIVE, true);
+            KRATOS_INFO("DetermineOpenPipingElements") << "tip_element " << tip_element << std::endl;
+
+            // Get all open pipe elements
+            std::function<bool(Element*)> filter = [](Element* i) {
+                return i->Has(PIPE_ACTIVE) && i->GetValue(PIPE_ACTIVE);
+            };
+            auto OpenPipeElements = rPipingElements | boost::adaptors::filtered(filter);
+            KRATOS_INFO_IF("PipingLoop", this->GetEchoLevel() > 0 && rank == 0)
+                << "Number of Open Pipe Elements: " << boost::size(OpenPipeElements) << std::endl;
+
+            // non-lin picard iteration, for deepening the pipe
+            // Todo JDN (20220817):: Deal with Equilibrium redundancy
+            // Equilibrium = check_pipe_equilibrium(OpenPipeElements, amax, mPipingIterations);
+            check_pipe_equilibrium(OpenPipeElements, MaxPipeHeight, mPipingIterations);
```

Comment:
To align with function name style guide:
```suggestion
            CheckPipeEquilibrium(OpenPipeElements, MaxPipeHeight, mPipingIterations);
```

Reply:
done